common/lib/xmodule/xmodule/js/src/sequence/display.js

```diff
@@ -0,0 +1,409 @@
+            /**
+            * check for the updated problems for given sequence position.
+            * params:
+            *   'position' can be any sequence position.
+            */
+            return this.updatedProblems[position] !== void 0;
+        };
+
+        Sequence.prototype.hookUpProgressEvent = function() {
+            return $('.problems-wrapper').bind('progressChanged', this.updateProgress);
+        };
+
+        Sequence.prototype.mergeProgress = function(p1, p2) {
+            // if either is 'NA', return the other one
+            var w1, w2;
+            if (p1 === 'NA') {
+                return p2;
+            }
+            if (p2 === 'NA') {
+                return p1;
+            }
+
+            // Both real progresses
+            if (p1 === 'done' && p2 === 'done') {
+                return 'done';
+            }
+
+            // not done, so if any progress on either, in_progress
+            w1 = p1 === 'done' || p1 === 'in_progress';
+            w2 = p2 === 'done' || p2 === 'in_progress';
+
+            if (w1 || w2) {
+                return 'in_progress';
+            }
+
+            return 'none';
+        };
+
+        Sequence.prototype.updateProgress = function() {
+            var newProgress = 'NA',
+                self = this;
+            $('.problems-wrapper').each(function() {
+                var progress = $(this).data('progress_status');
+                newProgress = self.mergeProgress(progress, newProgress);
+            });
+            this.progressTable[this.position] = newProgress;
+        };
+
+        Sequence.prototype.enableButton = function(buttonClass, buttonAction) {
+            return this.$(buttonClass)
+                .removeClass('disabled')
+                .removeAttr('disabled')
+                .click(buttonAction);
+        };
+
+        Sequence.prototype.disableButton = function(buttonClass) {
+            return this.$(buttonClass).addClass('disabled').attr('disabled', true);
+        };
+
+        Sequence.prototype.setButtonLabel = function(buttonClass, buttonLabel) {
+            return this.$(buttonClass + ' .sr').html(buttonLabel);
+        };
+
+        Sequence.prototype.updateButtonState = function(
+            buttonClass, buttonAction, actionLabelPrefix, isAtBoundary, boundaryUrl
+        ) {
+            var buttonLabel;
+            if (isAtBoundary && boundaryUrl === 'None') {
+                return this.disableButton(buttonClass);
+            } else {
+                buttonLabel = actionLabelPrefix + (isAtBoundary ? ' Subsection' : ' Unit');
+                this.setButtonLabel(buttonClass, buttonLabel);
+                return this.enableButton(buttonClass, buttonAction);
+            }
+        };
+
+        Sequence.prototype.toggleArrows = function() {
+            var isFirstTab, isLastTab, nextButtonClass, previousButtonClass;
+            this.$('.sequence-nav-button').unbind('click');
+            // previous button
```

Comment:
Somehow a lot of the comments got moved up a line from the code that they relate to.